checker/match_test.go

```diff
@@ -110,6 +110,33 @@ func TestMatchHTTPMethods(t *testing.T) {
 	}
 }
 
+// HTTP Paths clause with empty list will match any path.
+func TestMatchHTTPPaths(t *testing.T) {
+	testCases := []struct {
+		title   string
+		paths   []*proto.HTTPMatchPathMatchTypes
+		reqPath string
+		result  bool
+	}{
+		{"empty", []*proto.HTTPMatchPathMatchTypes{}, "/foo", true},
+		{"exact", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Exact{Exact: "/foo"}}}, "/foo", true},
+		{"prefix", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Prefix{Prefix: "/foo"}}}, "/foobar", true},
+		{"exact fail", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Exact{Exact: "/foo"}}}, "/joo", false},
```

Comment:
how about `reqPath` = `/foobar` so we test that exact matches don't behave like prefixes